src/dapp/libexec/dapp/dapp

```diff
@@ -25,6 +25,10 @@ fi
 
 if [[ $2 = --help ]]; then
   exec "${0##*/}" help -- "$1"
+elif [[ $DAPP_SOLC_VERSION ]]; then
+  v=$DAPP_SOLC_VERSION
+  unset DAPP_SOLC_VERSION
```

Comment:
Why do we unset it here? I think that if someone exports the variable, they probably want this to persist, no?

Reply:
Ah, that's why I said this was hacky. When it detects the environment variable, it calls `dapp---use solc:$v [and everything else you put as commands]`. This in turn calls `dapp---nix-run` with an nix override that sets solc as the version you specified. `dapp --nix-run` can be used to alter more than `solc` if you wished. So if we had multiple versions of `hevm` as we do `solc`, and we specified a way to read that version, we could run `dapp` with a different `hevm` than the default.

Reply:
I unset because next time it runs, it should not go through that path, instead through the path below it

Reply:
> I unset because next time it runs, it should not go through that path, instead through the path below it

Should it though? Wouldn't you expect the side-effects of the env var you set to persist until you unset it explicitly?

Basically:

* user runs `export DAPP_SOLC_VERSION=0.5.2`
* user runs `dapp test` - equivalent to `dapp --use solc:0.5.2 test`
* user runs `dapp test` again -- eq to `dapp --use solc:0.4.24 test`

Is this the behaviour you're after? If so, I don't think it's intuitive.

Or is the env var only being unset in a sub-shell?

Reply:
Only in a sub-shell.
```
export DAPP_SOLC_VERSION=0.4.25
dapp --version
dapp --version
```
Will correctly use `0.4.25` both times.